File: pset2/readability/readability.c
// includes
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// declare functions

int main(void)
{
    string text = get_string("Text: ");
    float letters = 0;
    float words = 1;
    float sentence = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        // count letters
        if (isalpha(text[i]) > 0)
        {
            letters = letters + 1;
        }
        // count words
        if ((isspace(text[i]) > 0 && isalpha(text[i + 1]) > 0) || (isspace(text[i]) > 0 && isalpha(text[i + 2]) > 0))
        {
            words = words + 1;
        }
        // count sentence
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentence = sentence + 1;
        }
    }
    float L = letters * 100 / words;
    float S = sentence * 100 / words;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.f\n", round(index));
    }
}

File: pset6/readability/readability.py
from cs50 import get_string


def index(text):  # index function
    # define variable
    letters = 0
    words = 1
    sentence = 0
    # create loop fo every char in string
    for i in range(0, len(text)):
        # counting letters
        if text[i].isalpha():
            letters += 1
        # counting words - if it's space and i+1 or i+2 is char, then it's a next word
        elif text[i] == " " and (text[i + 1].isalpha() or text[i + 2].isalpha()):
            words += 1
        # counting sentence - if it's one of (.?!) then it's a next sentence
        elif text[i] in [".", "?", "!"] and not text[i - 1] in [".", "?", "!"]:
            sentence += 1
        i += 1

    # index calcs
    return 0.0588 * (letters * 100 / words) - 0.296 * (sentence * 100 / words) - 15.8


def grade(index):  # grade function
    if index > 16:
        return "Grade 16+"
    elif index < 1:
        return "Before Grade 1"
    else:
        return f"Grade {round(index)}"


def main():  # define main function
    # get text from user
    text = get_string("Text: ")
    # print grade
    print(grade(index(text)))


main()